crates/catalog/sql/src/catalog.rs

```diff
@@ -0,0 +1,397 @@
+                Box::pin(async move {
+            sqlx::query(&format!("select table_namespace, table_name, metadata_location, previous_metadata_location from iceberg_tables where catalog_name = '{}' and table_namespace = '{}' and table_name = '{}';",&catalog_name,
+                    &namespace,
+                    &name)).fetch_one(&mut **txn).await
+        })}).await.map_err(from_sqlx_error)?;
+            let row = query_map(&row).map_err(from_sqlx_error)?;
+
+            row.metadata_location
+        };
+        let file = self.storage.new_input(&metadata_location)?;
+
+        let mut json = String::new();
+        file.reader().await?.read_to_string(&mut json).await?;
+
+        let metadata: TableMetadata = serde_json::from_str(&json)?;
+
+        self.cache
+            .insert(identifier.clone(), (metadata_location, metadata.clone()));
+
+        let table = Table::builder()
+            .file_io(self.storage.clone())
+            .identifier(identifier.clone())
+            .metadata(metadata)
+            .build();
+
+        Ok(table)
+    }
+
+    async fn create_table(
+        &self,
+        namespace: &NamespaceIdent,
+        creation: TableCreation,
+    ) -> Result<Table> {
+        let location = creation.location.ok_or(Error::new(
+            ErrorKind::DataInvalid,
+            "Table creation with the Sql catalog requires a location.",
+        ))?;
+
+        let uuid = Uuid::new_v4();
+        let metadata_location =
+            location.clone() + "/metadata/" + "0-" + &uuid.to_string() + ".metadata.json";
+
+        let metadata = TableMetadata::builder()
+            .location(location)
+            .current_schema_id(creation.schema.schema_id())
+            .last_column_id(creation.schema.highest_field_id())
+            .schemas(HashMap::from_iter(vec![(
+                creation.schema.schema_id(),
+                creation.schema.into(),
+            )]))
+            .partition_specs(HashMap::new())
+            .last_partition_id(0)
+            .current_snapshot_id(-1)
+            .sort_orders(HashMap::new())
+            .default_sort_order_id(0)
+            .build();
+
+        let file = self.storage.new_output(&metadata_location)?;
+        file.writer()
+            .await?
+            .write_all(&serde_json::to_vec(&metadata)?)
+            .await?;
+        {
+            let mut connection = self.connection.lock().await;
+            connection.transaction(|txn|{
+                let catalog_name = self.name.clone();
+                let namespace = namespace.encode_in_url();
+                let name = creation.name.clone();
+                let metadata_location = metadata_location.to_string();
+                Box::pin(async move {
+            sqlx::query(&format!("insert into iceberg_tables (catalog_name, table_namespace, table_name, metadata_location) values ('{}', '{}', '{}', '{}');",catalog_name,namespace,name, metadata_location)).execute(&mut **txn).await
```

Comment:
Would it not be preferable to use parameterized queries for all of the queries in here?

```rust
sqlx::query("insert into iceberg_tables (catalog_name, table_namespace, table_name, metadata_location) values (?, ?, ?, ?);")
.bind(catalog_name)
.bind(namespace)
.bind(name)
.bind(metadata_location)
.execute(&mut **txn).await
```

Reply:
Good idea, that's probably better

Reply:
👍🏻